Source/WebCore/editing/cocoa/DataDetection.mm

```diff
@@ -292,12 +290,11 @@ static void removeResultLinksFromAnchor(Element& element)
 
 static bool searchForLinkRemovingExistingDDLinks(Node& startNode, Node& endNode)
 {
-    for (Node* node = &startNode; node; node = NodeTraversal::next(*node)) {
-        if (is<HTMLAnchorElement>(*node)) {
-            auto& anchor = downcast<HTMLAnchorElement>(*node);
-            if (!equalLettersIgnoringASCIICase(anchor.attributeWithoutSynchronization(x_apple_data_detectorsAttr), "true"_s))
+    for (auto* node = &startNode; node; node = NodeTraversal::next(*node)) {
+        if (RefPtr anchor = dynamicDowncast<HTMLAnchorElement>(*node)) {
+            if (!equalLettersIgnoringASCIICase(anchor->attributeWithoutSynchronization(x_apple_data_detectorsAttr), "true"_s))
                 return true;
-            removeResultLinksFromAnchor(anchor);
+            removeResultLinksFromAnchor(anchor.releaseNonNull());
```

Comment:
removeResultLinksFromAnchor() takes an `Element&`, not a `Ref<Element>&&`. Therefore, `*anchor` would be just as efficient here and more concise.